src/navigation/wallet-connect/screens/WalletConnectHome.tsx

```diff
@@ -103,21 +105,49 @@ const WalletConnectHome = () => {
     },
   );
   const session: WalletConnect | undefined = wcConnector?.connector;
-  const requests: IWCRequest[] = useAppSelector(({WALLET_CONNECT}) => {
+  const _requests: IWCRequest[] = useAppSelector(({WALLET_CONNECT}) => {
     return WALLET_CONNECT.requests.filter(request => request.peerId === peerId);
   });
 
+  const [requests, setRequests] = useState<IWCRequest[]>();
+
+  const allKeys = useAppSelector(({WALLET}: RootState) => WALLET.keys);
+
+  const decodeData = async () => {
+    const promises = [];
+    for await (const r of _requests) {
+      const data = r.payload.params[0].data;
```

Comment:
@rastajpa Are you sure the method you are passing here is one that is unique to the MRC20 ABI? If not, there is a possibility that it will return as ERC20 as they share 7 methods in common.

Reply:
good catch! you're right, I've been testing and it's not possible, so this workaround will fail in some cases. I'm working on a new solution, thank you

Reply:
thanks @escottalexander, now I sent some changes with a better way to get chain. Changes in bitcore will not be necessary for the moment.